src/main/java/org/folio/circulation/domain/LoanService.java

```diff
@@ -30,24 +30,25 @@ public LoanService(Clients clients) {
     if(requests.isEmpty()) {
       return completedFuture(succeeded(records));
     }
-    /*
-      This gets the top request, since UpdateRequestQueue.java#L106 updates the request queue prior to loan creation.
-      If that sequence changes, the following will need to be updated to requests.stream().skip(1).findFirst().orElse(null)
-      and the condition above could do a > 1 comparison. (CIRC-277)
-    */
-    Request nextRequestInQueue = requests.stream().findFirst().orElse(null);
-    if (nextRequestInQueue == null || nextRequestInQueue.getRequestType() != RequestType.RECALL) {
+    
+    //check to see if there are any recalls in the Queue
+    Boolean isRecallInQueue = requests.stream().anyMatch(request -> request.getRequestType() == RequestType.RECALL);
+
+    //if no, exit 
+    if (!isRecallInQueue) {
       return completedFuture(succeeded(records));
     }
 
+    //if there are any recalls, check to see if the wasDueDateChangedByRecall flag on the loan is already set
+
     final Loan loanToRecall = records.getLoan();
     final LoanPolicy loanPolicy = loanToRecall.getLoanPolicy();
 
-    // We don't need to apply the recall
+    // if it is, we don't need to apply the recall
     if (loanToRecall.wasDueDateChangedByRecall()) {
       return completedFuture(succeeded(records));
     }
-
+    //otherwise, apply the recall 
```

Comment:
I think this comment is also redundant.

I'm aware the code in this class had lots of comments and so you may well be trying to stick to that. In general I'd prefer to avoid them. If we don't think the code is expressive enough, I'd prefer we improved the code rather than added comments.